src/analysis/complex/isometry.lean

```diff
@@ -0,0 +1,185 @@
+    h₂, add_left_cancel_iff, mul_self_eq_mul_self_iff] at h₁,
+end
+
+lemma linear_isometry.l0 {f : ℂ →ₗᵢ[ℝ] ℂ} (h : f 1 = 1) (z : ℂ) : ∥f z - 1∥ = ∥z - 1∥ := begin
+  rw ←linear_isometry.norm_map f (z - 1),
+  rw linear_isometry.map_sub,
+  rw h,
+end
+
+lemma linear_isometry.l1 {f : ℂ →ₗᵢ[ℝ] ℂ} (h : f 1 = 1) (z : ℂ) : z + conj z = f z + conj (f z) :=
+begin
+  have := linear_isometry.l0 h z,
+  apply_fun λ x, x ^ 2 at this,
+  simp only [norm_eq_abs, ←norm_sq_eq_abs] at this,
+  rw [←of_real_inj, ←mul_conj, ←mul_conj] at this,
+  rw [conj.map_sub, conj.map_sub] at this,
+  simp only [sub_mul, mul_sub, one_mul, mul_one] at this,
+  rw [mul_conj, norm_sq_eq_abs, ←norm_eq_abs, linear_isometry.norm_map] at this,
+  rw [mul_conj, norm_sq_eq_abs, ←norm_eq_abs] at this,
+  simp only [sub_sub, sub_right_inj, mul_one, of_real_pow, ring_hom.map_one, norm_eq_abs] at this,
+  simp only [add_sub, sub_left_inj] at this,
+  rw [add_comm, ←this, add_comm],
+end
+
+lemma linear_isometry.re {f : ℂ →ₗᵢ[ℝ] ℂ} (h : f 1 = 1) (z : ℂ) : (f z).re = z.re := begin
+  apply hf_re,
+  intro z,
+  apply linear_isometry.l1 h,
+end
+
+lemma linear_isometry_complex_aux (f : ℂ →ₗᵢ[ℝ] ℂ) (h : f 1 = 1) :
+  (∀ z, f z = z) ∨ (∀ z, f z = conj z) := by {
+  have h0 : f I = I ∨ f I = -I,
+  { have : |f I| = 1,
+    { rw [←norm_eq_abs, linear_isometry.norm_map, norm_eq_abs, abs_I] },
+    simp only [ext_iff, ←and_or_distrib_left, neg_re, I_re, neg_im, neg_zero],
+    split,
+    { rw ←I_re,
+      rw linear_isometry.re h },
+    { apply hf_im,
+      { intro z, rw [←norm_eq_abs, ←norm_eq_abs, linear_isometry.norm_map] },
+      { intro z, rw linear_isometry.re h, } } },
+  refine or.imp (λ h1, _) (λ h1 z, _) h0,
+  { suffices : f.to_linear_map = linear_isometry.id.to_linear_map,
+    { simp [this, ←linear_isometry.coe_to_linear_map, linear_map.id_apply] },
+    apply is_basis.ext is_basis_one_I,
+    intro i,
+    fin_cases i,
+    { simp [h] },
+    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
+        linear_isometry.id_to_linear_map, linear_map.id_coe, id.def, matrix.cons_val_one],
+      exact h1, } },
+  { suffices : f.to_linear_map = conj_li.to_linear_map,
+    { rw [←linear_isometry.coe_to_linear_map, this],
+      simp only [linear_isometry.coe_to_linear_map],
+      refl },
+    apply is_basis.ext is_basis_one_I,
+    intro i,
+    fin_cases i,
+    { simp only [h, linear_isometry.coe_to_linear_map, matrix.cons_val_zero],
+      change 1 = conj 1,
+      ext; simp },
+    { simp only [matrix.head_cons, linear_isometry.coe_to_linear_map,
+        linear_isometry.id_to_linear_map, linear_map.id_coe, id.def, matrix.cons_val_one],
+      change f I = conj I,
+      rw conj_I,
+      exact h1, } },
+}
+
+lemma linear_isometry_complex (f : ℂ →ₗᵢ[ℝ] ℂ) :
+  ∃ a : ℂ, |a| = 1 ∧ ((∀ z, f z = a * z) ∨ (∀ z, f z = a * conj z)) := by {
+  let a := f 1,
+  use a,
+  split,
+    {
```

Comment:
Braces are not alone on their line (see https://leanprover-community.github.io/contribute/style.html for style examples).